docs/architecture/adr-011-optimistic-blob-size-independent-inclusion-proofs-and-pfd-fraud-proofs.md

```diff
@@ -0,0 +1,173 @@
+# ADR 011: Optimistic Blob Size Independent Inclusion Proofs and PFB Fraud Proofs
+
+## Changelog
+
+- 18.11.2022: Initial Draft
+
+## Context
+
+The blob inclusion verification game is between the verifier and a prover. The verifier has the blob or its commitment and access to the block header of Celestia. The prover wants to prove the inclusion of the blob in the DA Layer.
+
+Validators check if the PFB commitment matches the commitment that is referenced in [ProcessProposal](https://github.com/celestiaorg/celestia-app/blob/3473000a9ff04fccfbba83929711fe11643b782c/app/process_proposal.go#L113). Given that 2/3 of validators can collude and break this consensus rule, we want a fraud-proof to inform light clients of violations of this rule. If we assume that we have this fraud-proof then we can use the check in `ProcessProposal` to our advantage for an optimistic blob size independent inclusion proof.
+
+## Blob size independent inclusion proof - PFB inclusion proof
+
+Instead of proving the Merkle proof over all subtree roots over the blob we create a Merkle proof over the PFB transaction as it includes the commitment already. Therefore this proof will include the shares where the PFB transaction lies in the Celestia square plus the inclusion proof of those shares to the `DataRoot`.
+
+The verifier could do the following steps:
+
+- Verify that the namespaceID is the transaction namespaceID (i.e. `namespace.ID{0, 0, 0, 0, 0, 0, 0, 1}`). We have to do this so we don't accidentally verify a blob share that looks like a PFB transaction
+- Unserialize the PFB transaction
+- Verify the signatures
```

Comment:
[question] are there multiple signatures? I assume there would only be one on a PFB transaction

```suggestion
- Verify that the PFB transaction signature is valid
```

Reply:
You can very different signature schemes. This is why a PFB can grow to so many bytes